[New KB Article] Why am I not receiving PR/MR comments? (#1069)
returntocorp/semgrep-docs: docs/kb/semgrep-cloud-platform/missing-pr-commments.md

---
description: Use this reference to check why you may not be receiving Semgrep comments on PRs/MRs.
tags:
  - Semgrep CI

>   - PR Comments

Reviewer: ```suggestion
  - PR comments
```

>   - Rule Board

Reviewer: ```suggestion
  - Rule management
```
This feature's name is tricky. It's called Rule Management (capital M) in the header within the product, then "Rule" and "Board" in the left sidenav within the product, and it's a [T3 feature in product marketing docs](https://www.notion.so/semgrep/Product-and-feature-naming-conventions-16c6cf54cc05479ea85e1b10597a4147), which means that following convention, it should be "Rule management" and not "Rule Management" when it's being marketed. I apologize for the confusion.
<img width="942" alt="Snag_ac5649" src="https://github.com/returntocorp/semgrep-docs/assets/20766840/2a559440-5b37-45ed-9973-099c61bb57d2">

---

> # Why am I not receiving PR/MR comments?

Reviewer: ```suggestion
# Why am I not receiving PR or MR comments?
```

We reserve slashes for filenames and code ([Slashes in GSG](https://developers.google.com/style/slashes)) 

This is one of those things a linter like Semgrep or Vale would be great to catch :)

If you have configured Semgrep in CI and Semgrep Cloud Platform to create comments when a rule generates a finding in a PR or MR, but you are not seeing those comments, check the following possibilities.

> ## Are comments supported on your SCM?

Reviewer: ```suggestion
## Are comments supported in your source code manager (SCM)?
```
Spell out SCM at first use :)

> PR/MR Comments are currently supported for GitHub, GitLab, and Bitbucket. If you are using another SCM, comments are not yet supported. Bitbucket Data Center is also not supported yet.

Reviewer: ```suggestion
PR or MR comments are currently supported for:

* All GitHub plans
* All GitLab plans
* Bitbucket Cloud

PR or MR comments are not supported for:

* Bitbucket Data Center
* Azure Repos
* Any other SCM or repository provider
```

Suggesting to make it a list to make it easy to skim.

I suspect some variant of this information may live in the CI docs, so after you merge this I may turn this into a react component so we can import it (more a note for myself)

Author: yeah this is a good change, thank you!

> If you are using a self-hosted version of GitHub (GitHub Enterprise) or GitLab (GitLab Self-Managed), see [Integrating Semgrep Cloud Platform with self-hosted enterprise repositories](/docs/semgrep-cloud-platform/scm/) for more details on configuration.

Reviewer: Thank you for this.

## Have you configured permissions and tokens correctly?

### GitHub

> GitHub relies on the Semgrep GitHub app to make comments on code. To receive comments on a project, you must [onboard](/docs/semgrep-code/getting-started/#option-b-adding-a-repository-from-github-gitlab-or-bitbucket) the project to Semgrep Cloud Platform, or ensure that your GitHub app is configured with permissions for all repositories. See [Enabling GitHub pull request comments](/docs/semgrep-cloud-platform/github-pr-comments/) for details.

Reviewer: ```suggestion
GitHub relies on the Semgrep GitHub app to make comments on code. To receive comments on a project, ensure that you have performed the following steps:

* You have [onboarded](/docs/semgrep-code/getting-started/#option-b-adding-a-repository-from-github-gitlab-or-bitbucket) the project to Semgrep Cloud Platform.
* You have configured your GitHub app with permissions for all repositories that are scanned by Semgrep Cloud Platform. See [Enabling GitHub pull request comments](/docs/semgrep-cloud-platform/github-pr-comments/) for details.
```

Just making it consistent with the list below.
Also (I am so sorry) GSG prefers a more explicit style of [writing introductory paragraphs to procedures](https://developers.google.com/style/procedures), so I suggested a complete sentence.

### GitLab and Bitbucket

> For MR comments on GitLab and PR comments on Bitbucket, you must:

Reviewer: ```suggestion
For MR comments on GitLab and PR comments on Bitbucket, ensure that you have performed the following steps:
```

> * configure a token with appropriate permissions to comment on MRs or PRs

Reviewer: ```suggestion
* Configure a token with appropriate permissions to comment on MRs or PRs.
```

Author: ```suggestion
* You have configured a token with appropriate permissions to comment on MRs or PRs.
```

> * add the token as a variable in your project or workspace configuration

Reviewer: ```suggestion
* You have added the token as a variable in your project or workspace configuration
```

> * add the token value to your CI configuration

Reviewer: ```suggestion
* You have added the token value to your CI configuration.
```

> The GitLab token should have `api` scope and be added to the project's CI/CD settings. See [Enabling GitLab merge request comments](/docs/semgrep-cloud-platform/gitlab-mr-comments/) for details.

Reviewer: So - the tricky thing about combining the two (GitLab and Bitbucket) is that these 2 paragraphs, lines 33 and 34, are actually for 2 separate audiences. You may want to separate the sections. I'm sorry I couldn't help restructure this thing entirely, but if you want I can push a commit, just let me know.

Author: Is it okay to make two H4 sections here? I'm definitely open to that, but I did like highlighting that the outline is similar for both.

Author: That's what I went to for now, but we can definitely iterate later :)

The Bitbucket token should be a repository access token (or workspace access token, for Bitbucket Cloud Premium only). See [Enabling Bitbucket pull request comments](/docs/semgrep-cloud-platform/bitbucket-pr-comments/) for details.

## Have you placed the rule in Comment or Block?

> In order to receive comments, the rule must be placed in a policy that will generate comments: Comment or Block. If the rule is in a Monitor policy, it will not generate comments.

Reviewer: ```suggestion
In order to receive comments, the rule must be placed in a policy that can generate comments:
* Comment
* Block.

If the rule is in a Monitor policy, it does not generate comments.
```

Yet Another List. I knowwwwww.

Here is the style guide citation if you're not already sick of seeing them: [Present tense](https://developers.google.com/style/tense).

Author: Present tense definitely, but I am not sure about the use of a list here. I guess it is more future-proof, in case we have more policies later. :)

> ## Is this the first time this finding has been identified?

Reviewer: This is a very good catch. We all know this at the back of our heads and take it for granted when it comes to external users.

> PR/MR comments are generated when a finding is new. If a finding is [not new](/docs/semgrep-code/findings/#deduplicating-findings) (if it was seen in a previous Semgrep [scan](/docs/semgrep-code/getting-started/#performing-a-scan)) then a comment will not be generated.

Reviewer: ```suggestion
PR or MR comments are generated when a finding is new. If a finding is was seen in a previous scan, it is [not new](/docs/semgrep-code/findings/#deduplicating-findings) and a comment is not generated.
```

This is mostly a style thing IMO, so feel free to ignore completely.
* GSG prefers to avoid [parenthesis as "some" people skip them](https://developers.google.com/style/parentheses)
* I suggest (very mildly) to remove the scan link as they've come so far down the doc they are likely to already know how to scan.

Author: yeah fair, they probably know what "scan" means & how to do it :)

This prevents repeated comments on findings that have already notified developers.

## If you're still having trouble

> If you've addressed these issues but are still not seeing comments, please [reach out for help](docs/support/), and provide information on what you've tried so far.

Reviewer: I sure hope our readers remember that last bit!

TY again for making it this far.
